File: neo4j/sample.py
from neo4j import GraphDatabase

uri = "neo4j://localhost:7687"
driver = GraphDatabase.driver(uri, auth=("neo4j", 'password'))

def create_friend_of(tx, name, friend):
    tx.run("create (a:Person)-[:KNOWS]->(f:Person {name: $friend})"
           "where a.name = $name "
           "return f.name as friend", name=name, friend=friend)

with driver.session() as session:
    session.write_transaction(create_friend_of, "Alice", "Bob")

with driver.session() as session:
    session.write_transaction(create_friend_of, "Alice", "Carl")

driver.close()

File: neo4j/sample.java
Driver driver = GraphDatabase.driver("neo4j://localhost:7687", AuthTokens.basic("jim", "1234"))

try( Session session = driver.session() )
{
    session.writeTransaction(new TransactionWork<String>()
    {
        @Override
        public String execute( Transaction transaction)
        {
            Result result = 
                transaction.run(
                    "CREATE (a:Person {name: $name}) " +
                    "-[:KNOWS]->(f:Person {name: $friend}) " +
                    "RETURN f.name as friend",
                    parameters("name", "Alice", "friend", "Bob")
                );

            return result.single().get(0).asString();
        }
    } );
}

driver.close();